src/path/sentry_path_unix.c

```diff
@@ -501,3 +501,61 @@ sentry__path_append_buffer(
     return write_buffer_with_flags(
         path, buf, buf_len, O_RDWR | O_CREAT | O_APPEND);
 }
+
+struct sentry_filewriter_s {
+    size_t byte_count;
+    int fd;
+};
+
+MUST_USE sentry_filewriter_t *
+sentry__filewriter_new(const sentry_path_t *path)
+{
+    int fd = open(path->path, O_RDWR | O_CREAT | O_TRUNC,
+        S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH);
+    if (fd < 0) {
+        return NULL;
+    }
+    sentry_filewriter_t *result = SENTRY_MAKE(sentry_filewriter_t);
```

Comment:
there should probably be error handling for a possible failure here.

Reply:
Yes, they certainly should 🤦. Thx!